scripts/post-processing/3-cluster_analysis.R

```diff
@@ -0,0 +1,158 @@
+#        directory
+# "-l" : Optional label for output files. 
+
+# Command line example:
+#
+# Rscript scripts/7-cluster_analysis.R \
+# -d pca_data \
+# -m metadata.tsv \
+# -o results \
+# -l darmanis
+# 
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Import functions for this analysis
+source(file.path("scripts", "util", "clustering_statistics_functions.R"))
+
+# ggplot2 library
+library(ggplot2)
+library(optparse)
+
+#--------------------------------Set up options--------------------------------#
+option_list <- list(
+  make_option(opt_str = c("-d", "--data"), type = "character", default = getwd(),
+              help = "Directory path to folder of dataset(s) of gene matrices 
+              (probably dimension reduced) in tsv format, gene x sample to test.",
+              metavar = "character"),
+  make_option(opt_str = c("-m", "--metadata"), type = "character",
+              default = "none", help = "Path to metadata file that contains 
+              only the metadata variables you wish to test and label by.",
+              metavar = "character"),
+  make_option(opt_str = c("-o", "--output"), type = "character",
+              default = getwd(), help = "Directory where you would like the
+              output to go", metavar = "character"),
+  make_option(opt_str = c("-l", "--label"), type = "character",
+              default = "", help = "Optional label for output files",
+              metavar = "character")
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# Create the output for results folder if it does not exist
+if (!dir.exists(opt$output)) {
+  message(paste("Output folder:", opt$output, "does not exist, creating one"))
+  dir.create(opt$output)
+}
+
+# Append an underscore to the label if it is given
+if (opt$label != "") {
+  opt$label <- paste0("_", opt$label)
+}
+
+#---------------------------------Read in data---------------------------------#
+# Obtain list of file names: 
+if (dir.exists(opt$data)) { # If directory is given, read all tsv files
+  # Get the file names of all the normalized files
+  dataset.files <- dir(opt$data, full.names = TRUE)
+
+  # Only read in tsv files
+  dataset.files <- grep("\\.tsv$", dataset.files, value = TRUE)
+  
+} else { # Otherwise, make file list by separating by spaces
+  dataset.files <- opt$data <- strsplit(opt$data, " ")
+}
+
+# Read in each of the normalization files
+datasets <- lapply(dataset.files, readr::read_tsv)
+
+# Obtain names of datasets
+dataset.names <- stringr::word(dataset.files, sep = "/", -1)
+
+# Remove file extension name and make it the name in the list
+names(datasets) <- gsub("\\..*$", "", dataset.names)
+
+#-------------------------------Set up metadata--------------------------------#
+# Read in the metadata
+meta <- as.list(readr::read_tsv(opt$metadata))
+
+# Make all the metadata into factors
+meta <- lapply(meta, as.factor)
+
+# Obtain variable names from metadata import
```

Comment:
Add a brief note about why these names are structured this way (i.e., how these were obtained from GEO), please